src/ucp/wireup/wireup.c

```diff
@@ -100,11 +100,13 @@ static ucp_lane_index_t ucp_wireup_get_msg_lane(ucp_ep_h ep, uint8_t msg_type)
 
 ucs_status_t ucp_wireup_msg_progress(uct_pending_req_t *self)
 {
-    ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
-    ucp_ep_h ep        = req->send.ep;
+    ucp_request_t *req  = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_ep_h ep         = req->send.ep;
+    ucs_status_t status = UCS_OK;
     ssize_t packed_len;
     unsigned am_flags;
-    ucs_status_t status;
+
+    UCS_ASYNC_BLOCK(&ep->worker->async);
```

Comment:
what does it protect? can you please give potential raise scenario?
async is possible for UD, where UCT level is protected

Reply:
CM disconnected event which adds `UCP_EP_FLAG_DISCONNECT_CB_CALLED`, i.e. `ucp_cm_disconnect_cb()`
or `ucp_worker_set_ep_failed()` which adds `UCP_EP_FLAG_FAILED` flag

Reply:
CM disconnect and set_ep_failed could be called in async thread whereas sending WIREUP MSG is done in the main thread

Reply:
which data fields does this lock protect?

Reply:
`ep->flags`